File: zkext/trunk/src/com/ff/Component.java
package com.ff;

import com.ff.annotation.Property;
import com.ff.enums.UpdateType;
import com.ff.event.PropertyChangeEvent;


public class Component extends AbstractComponent{

	static {
		registerEvent(PropertyChangeEvent.class, Component.class, PropertyChangeEvent.NAME, CE_IMPORTANT);
		//ACTIVATE
		//ADDED
		//AFTERRENDER
		//BEFOREACTIVATE
		//BEFOREDEACTIVATE
		//BEFOREDESTROY
		//BEFOREHIDE
		//BEFORERENDER
		//BEFORESHOW
		//DEACTIVATE
		//DESTROY
		//DISABLE
		//ENABLE
		//HIDE
		//MOVE
		//REMOVED
		//RENDER
		//RESIZE
		//SHOW
	}	
		
	@Property
	String border;
	@Property
	Boolean showBorder;
	@Property 
	String cls;	
	@Property 
	Boolean disabled;
	@Property 
	String disabledCls;
	@Property 
	Boolean draggable;	
	@Property 
	Integer height;
	@Property 
	Boolean hidden;
	@Property 
	String hideMode;		
	@Property 
	String margin;
	@Property 
	Integer maxHeight;
	@Property 
	Integer maxWidth;
	@Property 
	Integer minHeight;
	@Property 
	Integer minWidth;
	@Property 
	String overCls;
	@Property 
	String padding;				
	@Property 
	String style;		
	@Property(updateServer=UpdateType.ALWAYS)
	Integer width;
	@Property
	String region;
	@Property
	String styleWidth;
	@Property
	String styleHeight;
	@Property
	Integer flex;
	@Property 
	String anchor;
	@Property 
	Float columnWidth;
	
	
	public void show(){
		callMethod("show", null);
	}
	
	
	
	public String getBorder() {
		return border;
	}
	public void setBorder(String border) {
		this.border = border;
	}
	public String getCls() {
		return cls;
	}
	public void setCls(String cls) {
		this.cls = cls;
	}
	public Boolean getDisabled() {
		return disabled;
	}
	public void setDisabled(Boolean disabled) {
		this.disabled = disabled;
	}
	public String getDisabledCls() {
		return disabledCls;
	}
	public void setDisabledCls(String disabledCls) {
		this.disabledCls = disabledCls;
	}
	public Boolean getDraggable() {
		return draggable;
	}
	public void setDraggable(Boolean draggable) {
		this.draggable = draggable;
	}
	public Integer getHeight() {
		return height;
	}
	public void setHeight(Integer height) {
		this.height = height;
	}
	public Boolean getHidden() {
		return hidden;
	}
	public void setHidden(Boolean hidden) {
		this.hidden = hidden;
	}
	public String getHideMode() {
		return hideMode;
	}
	public void setHideMode(String hideMode) {
		this.hideMode = hideMode;
	}
	public String getMargin() {
		return margin;
	}
	public void setMargin(String margin) {
		this.margin = margin;
	}
	public Integer getMaxHeight() {
		return maxHeight;
	}
	public void setMaxHeight(Integer maxHeight) {
		this.maxHeight = maxHeight;
	}
	public Integer getMaxWidth() {
		return maxWidth;
	}
	public void setMaxWidth(Integer maxWidth) {
		this.maxWidth = maxWidth;
	}
	public Integer getMinHeight() {
		return minHeight;
	}
	public void setMinHeight(Integer minHeight) {
		this.minHeight = minHeight;
	}
	public Integer getMinWidth() {
		return minWidth;
	}
	public void setMinWidth(Integer minWidth) {
		this.minWidth = minWidth;
	}
	public String getOverCls() {
		return overCls;
	}
	public void setOverCls(String overCls) {
		this.overCls = overCls;
	}
	public String getPadding() {
		return padding;
	}
	public void setPadding(String padding) {
		this.padding = padding;
	}
	public String getStyle() {
		return style;
	}
	public void setStyle(String style) {
		this.style = style;
	}
	public Integer getWidth() {
		return width;
	}
	public void setWidth(Integer width) {
		this.width = width;		
	}
	public String getRegion() {
		return region;
	}
	public void setRegion(String region) {
		this.region = region;
	}
	public String getStyleWidth() {
		return styleWidth;
	}
	public void setStyleWidth(String styleWidth) {
		this.styleWidth = styleWidth;
	}
	public String getStyleHeight() {
		return styleHeight;
	}
	public void setStyleHeight(String styleHeight) {
		this.styleHeight = styleHeight;
	}
	public Integer getFlex() {
		return flex;
	}
	public void setFlex(Integer flex) {
		this.flex = flex;
	}
	public String getAnchor() {
		return anchor;
	}
	public void setAnchor(String anchor) {
		this.anchor = anchor;
	}
	public Float getColumnWidth() {
		return columnWidth;
	}
	public void setColumnWidth(Float columnWidth) {
		this.columnWidth = columnWidth;
	}



	public Boolean getShowBorder() {
		return showBorder;
	}



	public void setShowBorder(Boolean showBorder) {
		this.showBorder = showBorder;
	}
	   
	
}

File: zkext/trunk/src/web/js/zkext/ui/Component.js
/* Component.js*/

zkext.ui.Component = zk.$extends(zkext.ui.AbstractComponent,{	 
	$define: {	 		
		border:function(val){
			this.setProperty("border","setBorder",val);	
		},
		showBorder:function(val){
			this.setProperty("border",null,val);
		},
		cls:function(val){
			
		},
		disabled:function(val){
			
		},
		disabledCls:function(val){
			
		},
		draggable:function(val){
			
		},
		height:function(val){			
			//this.ext_.setHeight(val);		 
			this.setProperty("height","setHeight",val);
		},
		hidden:function(val){
			
		},
		hideMode:function(val){
			
		},
		margin:function(val){
			
		},
		maxHeight:function(val){
			
		},
		maxWidth:function(val){
			
		},
		minHeight:function(val){
			
		},
		minWidth:function(val){
			
		},
		overCls:function(val){
			
		},
		padding:function(val){
			
		},
		style:function(val){
			
		},
		width:function(val){			
			this.setProperty("width","setWidth",val);
			//this.ext_.setWidth(val);			
		},
		region:function(val){
			//alert(val);
			this.setProperty("region",null,val);
		},
		styleWidth:function(val){			
			this.setProperty("width","setWidth",val);		
		},
		styleHeight:function(val){			
			this.setProperty("height","setHeight",val);		
		},
		flex:function(val){
			this.setProperty("flex",null,val)
		},
		anchor:function(val){
			this.setProperty("anchor",null,val);
		},
		columnWidth:function(val){
			this.setProperty("columnWidth",null,val);
		}
	},
	
	
	show:function(){
		this.ext_.show();
	}
	
});